apps/hash-frontend/src/shared/notifications-context.tsx

```diff
@@ -115,64 +120,98 @@ export const NotificationsContextProvider: FunctionComponent<
                 {
-                  equal: [
-                    {
-                      path: [
-                        "properties",
-                        extractBaseUrl(
-                          types.propertyType.archived.propertyTypeId,
-                        ),
-                      ],
-                    },
-                    // @ts-expect-error -- We need to update the type definition of `EntityStructuralQuery` to allow for this
-                    //   @see https://linear.app/hash/issue/H-1207
-                    null,
+                  path: [
+                    "properties",
+                    extractBaseUrl(types.propertyType.archived.propertyTypeId),
                   ],
                 },
+                // @ts-expect-error -- We need to update the type definition of `EntityStructuralQuery` to allow for this
+                //   @see https://linear.app/hash/issue/H-1207
+                null,
+              ],
+            },
+            {
+              equal: [
                 {
-                  equal: [
-                    {
-                      path: [
-                        "properties",
-                        extractBaseUrl(
-                          types.propertyType.archived.propertyTypeId,
-                        ),
-                      ],
-                    },
-                    { parameter: false },
+                  path: [
+                    "properties",
+                    extractBaseUrl(types.propertyType.archived.propertyTypeId),
                   ],
                 },
+                { parameter: false },
               ],
             },
           ],
         },
+      ],
+    }),
+    [authenticatedUser],
+  );
+
+  const {
+    data: notificationRevisionsData,
+    loading: loadingNotificationRevisions,
+    refetch: refetchNotificationRevisions,
+  } = useQuery<
+    StructuralQueryEntitiesQuery,
+    StructuralQueryEntitiesQueryVariables
+  >(structuralQueryEntitiesQuery, {
+    pollInterval: 5_000,
+    variables: {
+      includePermissions: false,
+      query: {
+        filter: getNotificationsQueryFilter,
+        graphResolveDepths: zeroedGraphResolveDepths,
+        /**
+         * We need to obtain all revisions of the notifications
+         * to determine when they were created.
+         */
+        temporalAxes: {
+          pinned: { axis: "transactionTime", timestamp: null },
+          variable: {
+            axis: "decisionTime",
+            interval: { start: { kind: "unbounded" }, end: null },
+          },
+        },
+      },
+    },
+    skip: !authenticatedUser,
+    fetchPolicy: "cache-and-network",
+  });
+
+  const {
+    data: notificationsWithOutgoingLinksData,
+    loading: loadingNotificationsWithOutgoingLinks,
+    refetch: refetchNotificationsWithOutgoingLinks,
+  } = useQuery<
+    StructuralQueryEntitiesQuery,
+    StructuralQueryEntitiesQueryVariables
+  >(structuralQueryEntitiesQuery, {
+    pollInterval: 5_000,
```

Comment:
Can we assign this to a variable and share it between the two? It will crash if they become out of sync

Reply:
Addressed in https://github.com/hashintel/hash/pull/3511/commits/3051e5f93cd658f38b270e41c7425f05294bfcbe